### PR TITLE
[3.0][HttpKernel] remove deprecated functions and classes

### PR DESCRIPTION
**Pull request by vincentaubert**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by vincentaubert** [comment]
Hello,

I have a problem with the dependency injection of the Framework bundle.

in the services.xml file in the framework bundle I got this : 

<service id="http_kernel" class="Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel">
            <argument type="service" id="event_dispatcher" />
            <argument type="service" id="service_container" />
            <argument type="service" id="controller_resolver" />
            <argument type="service" id="request_stack" />
        </service>

But the Symfony\Component\HttpKernel\DependencyInjection\ContainerAwareHttpKernel class is marked as deprecated in the HttpKernel component.

By what should I replace this class to make it work ?

Thanks

**Comment by xabbuh** [comment]
@vincentaubert You can use `Symfony\Component\HttpKernel\HttpKernel` instead.

**Comment by dosten** [comment]
I'm not sure but i think you should replace it by `Symfony\Component\HttpKernel\HttpKernel` and remove `<argument type="service" id="service_container" />` from the definition.

**Comment by vincentaubert** [comment]
Thanks

File: src/Symfony/Bridge/Doctrine/Tests/Logger/DbalLoggerTest.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ public function testLogNonUtf8()
 
     public function testLogLongString()
     {
-        $logger = $this->getMock('Symfony\\Component\\HttpKernel\\Log\\LoggerInterface');
+        $logger = $this->getMock('Psr\\Log\\LoggerInterface');
```
Comment (fabpot) [inline src/Symfony/Bridge/Doctrine/Tests/Logger/DbalLoggerTest.php]: This change and the next one should happen in Symfony 2.7 instead.

Comment (vincentaubert) [inline src/Symfony/Bridge/Doctrine/Tests/Logger/DbalLoggerTest.php]: done in PR #14637

**Comment by vincentaubert** [comment]
ping @symfony/mergers

This PR is ready for merge.

The 2 failing tests have been fixed in the 2.7 branch via the #14637 PR but the 2.7 branch hasn't been merged in master at the moment.

I launched the tests locally with the patch and all the tests passed

**Comment by dosten** [comment]
You should remove this line https://github.com/vincentaubert/symfony/commit/d98acacadea23ece0683e1d6a5ba20935b279926#diff-bdd2593c9123e09266d82fcc955ba1c8R141

**Comment by dosten** [comment]
`ProfilerListener::getSubscribedEvents()` remove the `KernelEvents::REQUEST => array('onKernelRequest', 1024),` line and the comments above.

**Comment by vincentaubert** [comment]
fixed thanks

File: src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php [hunk 2]
```diff
@@ -200,10 +200,8 @@ public function getDumps($format, $maxDepthLimit = -1, $maxItemsPerDepth = -1)
         foreach ($this->data as $dump) {
             if (method_exists($dump['data'], 'withMaxDepth')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: `if` to be removed also

**Comment by vincentaubert** [comment]
done and code rebased on the latest master

File: src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php [hunk 3]
```diff
@@ -198,12 +198,6 @@ public function getDumps($format, $maxDepthLimit = -1, $maxItemsPerDepth = -1)
         $dumps = array();
 
         foreach ($this->data as $dump) {
-            if (method_exists($dump['data'], 'withMaxDepth')) {
-                $dumper->dump($dump['data']->withMaxDepth($maxDepthLimit)->withMaxItemsPerDepth($maxItemsPerDepth));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: You should keep this line!

Comment (vincentaubert) [inline src/Symfony/Component/HttpKernel/DataCollector/DumpDataCollector.php]: fixed

**Comment by fabpot** [comment]
Thank you @vincentaubert.

File: src/Symfony/Component/HttpKernel/EventListener/RouterListener.php [hunk 4]
```diff
@@ -67,34 +67,12 @@ public function __construct($matcher, RequestContext $context = null, LoggerInte
             throw new \InvalidArgumentException('You must either pass a RequestContext or the matcher must implement RequestContextAwareInterface.');
         }
 
-        if (!$requestStack instanceof RequestStack) {
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/EventListener/RouterListener.php]: requestStack must be made required

File: src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php [hunk 5]
```diff
@@ -49,13 +48,6 @@ class ProfilerListener implements EventSubscriberInterface
      */
     public function __construct(Profiler $profiler, RequestMatcherInterface $matcher = null, $onlyException = false, $onlyMasterRequests = false, RequestStack $requestStack = null)
```
Comment (Tobion) [inline src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php]: same problem

**Pull request merged**